src/packages/solid/utils/getWidthByUpcount.js

```diff
@@ -0,0 +1,20 @@
+export function getWidthByUpCount(theme, upCount = 1) {
```

Comment:
we need to add the license header here. this also should be written in TS or have a relevant d.ts file if that isn't possible